packages/backend/server/src/plugins/indexer/resolver.ts

```diff
@@ -0,0 +1,135 @@
+import { Args, Parent, ResolveField, Resolver } from '@nestjs/graphql';
+
+import { CurrentUser } from '../../core/auth';
+import { AccessController } from '../../core/permission';
+import { UserType } from '../../core/user';
+import { WorkspaceType } from '../../core/workspaces';
+import { Models } from '../../models';
+import { AggregateBucket } from './providers';
+import { IndexerService, SearchNodeWithMeta } from './service';
+import {
+  AggregateInput,
+  AggregateResultObjectType,
+  SearchInput,
+  SearchQueryOccur,
+  SearchQueryType,
+  SearchResultObjectType,
+} from './types';
+
+@Resolver(() => WorkspaceType)
+export class IndexerResolver {
+  constructor(
+    private readonly indexer: IndexerService,
+    private readonly ac: AccessController,
+    private readonly models: Models
+  ) {}
+
+  @ResolveField(() => SearchResultObjectType, {
+    description: 'Search a specific table',
+  })
+  async search(
+    @CurrentUser() me: UserType,
+    @Parent() workspace: WorkspaceType,
+    @Args('input') input: SearchInput
+  ): Promise<SearchResultObjectType> {
+    // currentUser can read the workspace
+    await this.ac.user(me.id).workspace(workspace.id).assert('Workspace.Read');
+    this.#addWorkspaceFilter(workspace, input);
+
+    const result = await this.indexer.search(input);
+    const nodes = await this.#filterUserReadableDocs(
+      workspace,
+      me,
+      result.nodes
+    );
+    return {
+      nodes,
+      pagination: {
+        count: result.total,
```

Comment:
use filtered total?

Reply:
The current response data structure follows the existing front-end indexer query data interface, so that the front-end refactoring can minimize the code change between local and cloud indexer.